ReactWindows/ReactNative.Shared/Bridge/ReactBridge.cs

```diff
@@ -108,11 +110,35 @@ public void Dispose()
 
         private void ProcessResponse(JToken response)
         {
+            ProcessResponse(response, false);
+        }
+
+        private void ProcessResponse(JToken response, bool isEndOfBatch)
+        {
+            bool localBatchHadNativeModuleCalls;
+
             if (response == null || response.Type == JTokenType.Null || response.Type == JTokenType.Undefined)
             {
+                if (isEndOfBatch)
+                {
+                    localBatchHadNativeModuleCalls = _batchHadNativeModuleCalls;
+
+                    _nativeModulesQueueThread.Dispatch(() =>
+                    {
+                        // the bridge could still be processing native calls when the idle signaler fires
+                        if (localBatchHadNativeModuleCalls)
+                        {
+                            _reactCallback.OnBatchComplete();
+                        }
+                    });
+
+                    _batchHadNativeModuleCalls = false;
+                }
+
```

Comment:
Would it be possible to move this logic into `ProcessResponseImmediate` (note rename comment above for `ProcessResponse(JToken)`)